src/sia.c

```diff
@@ -1,30 +1,42 @@
 #include "sia.h"
 
 #include <cx.h>
+#include <lib_standard_app/crypto_helpers.h>
 #include <os.h>
+#include <os_seed.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
 
 #include "blake2b.h"
 
+static void siaSetPath(uint32_t index, uint32_t path[static 5]) {
+    path[0] = 44 | 0x80000000;
+    path[1] = 93 | 0x80000000;
+    path[2] = index | 0x80000000;
+    path[3] = 0x80000000;
+    path[4] = 0x80000000;
+}
+
 void deriveSiaKeypair(uint32_t index, cx_ecfp_private_key_t *privateKey, cx_ecfp_public_key_t *publicKey) {
```

Comment:
Isn't this called only to retrieve the pubkey now that `deriveAndSign` do'nt use it anymore?
Then its prototype could be changed to only output the raw pubkey bytes taking a `uint8_t  raw_pubkey[static 65]` as parameter and calling directly `bip32_derive_with_seed_get_pubkey_256`.